### PR TITLE
2.x: cleanup, bugfixes, coverage 8/27-2

### PR DESCRIPTION
**Pull request by akarnokd**
- Remove unused code
- Improve coverage of `Single`, `Completable` and `Observable`
- Fix minor bugs in operators

### CODE REVIEW

File: src/main/java/io/reactivex/Observable.java [hunk 1]
```diff
@@ -1743,16 +1743,10 @@ public static int bufferSize() {
      */
     @SchedulerSupport(SchedulerSupport.CUSTOM)
     public static Observable<Long> interval(long initialDelay, long period, TimeUnit unit, Scheduler scheduler) {
-        if (initialDelay < 0) {
-            initialDelay = 0L;
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/Observable.java:1747]: Surprised we aren't throwing here (and others)

File: src/main/java/io/reactivex/Observable.java [hunk 2]
```diff
@@ -10108,7 +10082,7 @@ public final void subscribe(Observer<? super T> observer) {
     @SchedulerSupport(SchedulerSupport.NONE)
     public final Observable<T> take(long count) {
         if (count < 0) {
```
Comment (vanniktech) [inline src/main/java/io/reactivex/Observable.java]: also use verifyPositive here?

File: src/main/java/io/reactivex/Observable.java [hunk 3]
```diff
@@ -4701,12 +4685,8 @@ public final void blockingSubscribe(Observer<? super T> subscriber) {
...
     @SchedulerSupport(SchedulerSupport.NONE)
     public final <U extends Collection<? super T>> Observable<U> buffer(int count, int skip, Callable<U> bufferSupplier) {
-        if (count <= 0) {
-            throw new IllegalArgumentException("count > 0 required but it was " + count);
-        }
-        if (skip <= 0) {
-            throw new IllegalArgumentException("skip > 0 required but it was " + count);
-        }
+        verifyPositive(count, "count");
+        verifyPositive(skip, "skip");
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/Observable.java:4689]: I think one of these is wrong because `skip` can be 0 but `count` cannot.

Comment (JakeWharton) [inline src/main/java/io/reactivex/Observable.java:4689]: Based on other methods in this diff I would guess `verifyPositive` rejects 0

Comment (akarnokd) [reply src/main/java/io/reactivex/Observable.java:4689]: It creates buffers every `index % skip` so zero not allowed.

File: src/main/java/io/reactivex/observers/TestObserver.java [hunk 4]
```diff
@@ -895,15 +895,16 @@ private String fusionModeToString(int mode) {
      * @param time the waiting time
      * @param unit the time unit of the waiting time
      * @return this
-     * @throws InterruptedException if the wait is interrupted
+     * @throws RuntimeExceptio wrapping an InterruptedException if the wait is interrupted
```
Comment (JakeWharton) [inline src/main/java/io/reactivex/observers/TestObserver.java]: nit: missing 'n' on RuntimeException

**Comment by JakeWharton** [comment]
:+1:

**Pull request merged**